스프링 시큐리티 설정하기

workhardslave opened the pull request:
- 시큐리티 초기 설정 추가
- 시큐리티 설정 관련 테스트 코드 추가
- Member 클래스 및 테스트 코드 개선

workhardslave committed:
refactor: application.yml 파일 comment 추가

workhardslave committed:
Merge branch 'main' of https://github.com/MaetanTwosome/travel-reservation-service

workhardslave committed:
feat: 시큐리티 초기 설정 추가

workhardslave committed:
Merge branch 'main' of https://github.com/MaetanTwosome/travel-reservation-service into feature/login

workhardslave committed:
refactor: Member 클래스 및 테스트 코드 개선

- 필드 값으로 enum 타입 활용
- 각 필드 상세 조건 추가
- 회원 정보 수정 메서드 수정
- 중복 코드 제거

workhardslave committed:
test: 시큐리티 기본 설정 테스트 코드 추가

Closes #7

songjayhyun commented on src/main/java/maetanTwosome/trs/member/entity/Member.java:
```diff
@@ -2,29 +2,52 @@
+    private Role role;
+
+    @Enumerated(EnumType.STRING)
+    @Column
```
@Column 은 생략해도 될 것 같아용

workhardslave replied:
음 요거 nullable=false 설정 했어야했는데 빼먹었네요.
아래와 같이 수정할게용
```java
@Enumerated(EnumType.STRING)
@Column(nullable = false) // 추가
private Provider provider;

songjayhyun commented on src/test/java/maetanTwosome/trs/member/repository/MemberRepositoryTest.java:
```diff
@@ -1,87 +1,131 @@
+    @DisplayName("이메일 계정의 정보를 수정한다.")
     @Test
-    void update() {
+    void updateForEmailAccount() {
```
for는 생략해도 될 것 같네용

workhardslave replied:
다음과 같이 수정 의견 주신거죠?

as-is
updateForEmailAccount()
updateForOauthAccount()

to-be
updateEmailAccount()
updateOauthAccount()

songjayhyun replied:
녜녜

songjayhyun commented on src/test/java/maetanTwosome/trs/member/repository/MemberRepositoryTest.java:
```diff
@@ -1,87 +1,131 @@
+        assertThat(deleteMember).isNull();
+    }
+
+    private static Member getMemberByProvider(Provider provider) {
```
member를 만드는 내용이라 get은 좀 어색한 것 같아용. create어떤가요

songjayhyun commented on src/test/java/maetanTwosome/trs/config/SecurityConfigTest.java:
```diff
@@ -0,0 +1,35 @@
+
+@AutoConfigureMockMvc
+@SpringBootTest
+class SecurityConfigTest {
```
클래스 안에 주석으로 white list가 추가되면 좋을 것 같아용

workhardslave replied:
굿아이디어요 어디쪽에다 다는게 조을까여~

songjayhyun replied:
클래스 시작하는 부분이 좋을듯요

songjayhyun commented on src/main/java/maetanTwosome/trs/config/SecurityConfig.java:
```diff
@@ -0,0 +1,39 @@
+
+import java.util.stream.Stream;
+
+@EnableWebSecurity
```
스프링 부트 사용 시에, SecurityAutoConfiguration 에서 import 되는 WebSecurityEnablerConfiguration에 의해 자동으로 세팅되기 때문에 애노테이션 제거 할 수 있다는데, 확인 부탁드려용

workhardslave committed:
refactor: 코드 리뷰 반영

- Member 클래스 @Column 옵션 추가
- MemberRepositoryTest 테스트 코드 메서드명 수정
- 시큐리티
설정 파일 불필요한 어노테이션 제거
- SecurityConfigTest 주석 추가

Closes #7
